src/governance/LendingTermFactory.sol

```diff
@@ -104,77 +104,16 @@ contract LendingTermFactory is CoreRef {
             auctionHouses[auctionHouse],
             "LendingTermFactory: invalid auctionHouse"
         );
-        // must be an ERC20 (maybe, at least it prevents dumb input mistakes)
-        (bool success, bytes memory returned) = params.collateralToken.call(
-            abi.encodeWithSelector(IERC20.totalSupply.selector)
-        );
-        require(
-            success && returned.length == 32,
-            "LendingTermFactory: invalid collateralToken"
-        );
-
-        require(
-            params.maxDebtPerCollateralToken != 0, // must be able to mint non-zero debt
-            "LendingTermFactory: invalid maxDebtPerCollateralToken"
-        );
-
-        require(
-            params.interestRate < 1e18, // interest rate [0, 100[% APR
-            "LendingTermFactory: invalid interestRate"
-        );
-
-        require(
-            // 31557601 comes from the constant LendingTerm.YEAR() + 1
-            params.maxDelayBetweenPartialRepay < 31557601, // periodic payment every [0, 1 year]
-            "LendingTermFactory: invalid maxDelayBetweenPartialRepay"
-        );
-
-        require(
-            params.minPartialRepayPercent < 1e18, // periodic payment sizes [0, 100[%
-            "LendingTermFactory: invalid minPartialRepayPercent"
-        );
-
-        require(
-            params.openingFee <= 0.1e18, // open fee expected [0, 10]%
-            "LendingTermFactory: invalid openingFee"
-        );
-
-        require(
-            params.hardCap != 0, // non-zero hardcap
-            "LendingTermFactory: invalid hardCap"
-        );
-
-        // if one of the periodic payment parameter is used, both must be used
-        if (
-            params.minPartialRepayPercent != 0 ||
-            params.maxDelayBetweenPartialRepay != 0
-        ) {
-            require(
-                params.minPartialRepayPercent != 0 &&
-                    params.maxDelayBetweenPartialRepay != 0,
-                "LendingTermFactory: invalid periodic payment params"
-            );
-        }
```

Comment:
Because we don't know the parameters before being in the lending term contract, these tests are now in the LendingTerm initialize function